pennylane/gradients/finite_difference.py

```diff
@@ -347,6 +348,14 @@ def finite_diff(
         shapes.append(len(g_tapes))
 
     def processing_fn(results):
+        # Apply the same squeezing as in qml.QNode to make the transform output consistent.

```

Comment:
```suggestion
        # HOTFIX: Apply the same squeezing as in qml.QNode to make the transform output consistent.
```

I add this because I think, long term, we should avoid this behaviour altogether!